gremlinclient/client.py

```diff
@@ -153,9 +153,9 @@ def parser(f):
                     future.set_result(message)
                     self._closed = True
                 else:
+                    future.set_exception(RuntimeError(
+                        "{0} {1}".format(message.status_code, message.message)))
                     future.cancel()
```

Comment:
As it turns out, you can't really cancel Tornado's version of a future ``Future``. We wouldn't really want to cancel here anyway, since we have set an exception on the future.